scripts/background.js

```diff
@@ -145,85 +157,120 @@ function processHeaders(details) {
             }
         }
 
-        // 403 with the right header indicates a bypassable CAPTCHA
-        if (isBypassHeader(header) && details.statusCode == 403) {
-            doRedeem = true;
+        // correct status code with the right header indicates a bypassable Cloudflare CAPTCHA
+        if (isBypassHeader(header) && SPEND_STATUS_CODE.indexOf(details.statusCode) > -1) {
+            let iframe = (details.frameId > 0);
+            // check if the token should only be spent on an iframe
+            if (SPEND_IFRAME) {
+                if (!iframe) {
+                    activated = false;
+                } else {
+                    activated = true;
+                }
+            } else {
+                activated = true;
+            }
         }
     }
 
     // If we have tokens to spend, cancel the request and pass execution over to the token handler.
-    if (doRedeem && !spentUrl[url.href]) {
-        if (countStoredTokens() > 0) {
-            attemptRedeem(url, details.tabId);
-        } else {
-            // Store the url for redirection after captcha is solved
-            // Manual check for favicon urls
-            storedUrl = url.href;
-            let faviconIndex = storedUrl.indexOf("favicon");
-            if (faviconIndex != -1) {
-                storedUrl = storedUrl.substring(0, faviconIndex);
+    if (activated && !spentUrl[url.href]) {
+        let redeemOccurred = false;
+        if (DO_REDEEM) {
+            if (countStoredTokens() > 0 && activated) {
+                redeemOccurred = attemptRedeem(url, details.tabId);
+            } else if (countStoredTokens() == 0) {
+                // Update icon to show user that token may be spent here
```

Comment:
What happened to the logic for storing the url for redirection after the CAPTCHA is solved? Let me know if I'm missing something--was this logic moved elsewhere?

Reply:
It appeared that storedUrl was never actually used beyond it being set. I just decided to remove it.